builtin/providers/cloudstack/tags.go

```diff
@@ -0,0 +1,81 @@
+package cloudstack
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+// tagsSchema returns the schema to use for tags.
+//
+func tagsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Optional: true,
+	}
+}
+
+// setTags is a helper to set the tags for a resource. It expects the
+// tags field to be named "tags"
+func setTags(cs *cloudstack.CloudStackClient, d *schema.ResourceData, resourcetype string) error {
+
+	if d.HasChange("tags") {
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		create, remove := diffTags(tagsFromMap(o), tagsFromMap(n))
+		log.Printf("[DEBUG] tags to remove: %s", remove)
+		log.Printf("[DEBUG] tags to create: %s", create)
+
+		// Set tags
+		if len(remove) > 0 {
+			log.Printf("[DEBUG] Removing tags: %s from %s", remove, d.Id())
```

Comment:
I'm a little surprised, but just tested it and your right that `go vet` doesn't complain about this... Still I think `%v` is more inline with what you are trying to do here. If I see the line with the `%s` formatter I expect `remove` to be a string. But if you format with `%v` I expect it to be some type value...

Also looking at the docs, using `%v` seems to make much more sense:

```
%s	the uninterpreted bytes of the string or slice
%v	the value in a default format
```